src/main/java/es/deusto/spq/server/data/jdo/User.java

```diff
@@ -18,12 +19,16 @@
 	@Persistent(defaultFetchGroup="true")
 	private String password;
 	
+	/*@Persistent(defaultFetchGroup="true", mappedBy="user", dependentElement = "true")
+	private List<Review> reviews;*/
+	
 	public User(String userID, String name, String email, String password, String address) {
 		super();
 		this.userID = userID;
 		this.name = name;
 		this.email = email;
 		this.password = password;
+		//reviews = new ArrayList<Review>();
```

Comment:
Please remove unused code

Reply:
> A couple of things:
> 
> * There's a lot of uncommented code
> * Getting an error when running `ReviewDBTest`:
>   ```java
>   java.sql.SQLIntegrityConstraintViolationException: Duplicate entry '1' for key 'PRIMARY'
>   at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:117)
>   at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:97)
>   at com.mysql.cj.jdbc.exceptions.SQLExceptionsMapping.translateException(SQLExceptionsMapping.java:122)
>   at com.mysql.cj.jdbc.ClientPreparedStatement.executeInternal(ClientPreparedStatement.java:970)
>   at com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdateInternal(ClientPreparedStatement.java:1109)
>   at com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdateInternal(ClientPreparedStatement.java:1057)
>   at com.mysql.cj.jdbc.ClientPreparedStatement.executeLargeUpdate(ClientPreparedStatement.java:1377)
>   at com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdate(ClientPreparedStatement.java:1042)
>   at org.datanucleus.store.rdbms.datasource.dbcp2.DelegatingPreparedStatement.executeUpdate(DelegatingPreparedStatement.java:98)
>   at org.datanucleus.store.rdbms.datasource.dbcp2.DelegatingPreparedStatement.executeUpdate(DelegatingPreparedStatement.java:98)
>   at org.datanucleus.store.rdbms.ParamLoggingPreparedStatement.executeUpdate(ParamLoggingPreparedStatement.java:393)
>   at org.datanucleus.store.rdbms.SQLController.executeStatementUpdate(SQLController.java:427)
>   at org.datanucleus.store.rdbms.request.InsertRequest.execute(InsertRequest.java:414)
>   at org.datanucleus.store.rdbms.RDBMSPersistenceHandler.insertObjectInTable(RDBMSPersistenceHandler.java:162)
>   at org.datanucleus.store.rdbms.RDBMSPersistenceHandler.insertObject(RDBMSPersistenceHandler.java:138)
>   at org.datanucleus.state.StateManagerImpl.internalMakePersistent(StateManagerImpl.java:4569)
>   at org.datanucleus.state.StateManagerImpl.makePersistent(StateManagerImpl.java:4546)
>   at org.datanucleus.ExecutionContextImpl.persistObjectInternal(ExecutionContextImpl.java:2026)
>   at org.datanucleus.ExecutionContext.persistObjectInternal(ExecutionContext.java:339)
>   at org.datanucleus.store.types.SCOUtils.validateObjectForWriting(SCOUtils.java:1220)
>   at org.datanucleus.store.rdbms.scostore.ElementContainerStore.validateElementForWriting(ElementContainerStore.java:290)
>   at org.datanucleus.store.rdbms.scostore.FKListStore.validateElementForWriting(FKListStore.java:958)
>   at org.datanucleus.store.rdbms.scostore.FKListStore.internalAdd(FKListStore.java:562)
>   at org.datanucleus.store.rdbms.scostore.AbstractListStore.add(AbstractListStore.java:113)
>   at org.datanucleus.store.types.wrappers.backed.ArrayList.add(ArrayList.java:692)
>   at es.deusto.spq.server.data.jdo.Hotel.addReview(Hotel.java:39)
>   at es.deusto.spq.server.data.dao.HotelDAO.storeReview(HotelDAO.java:169)
>   at es.deusto.spq.ReviewDBTest.reviewDAOTest(ReviewDBTest.java:43)
>   at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
>   at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
>   at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
>   at java.base/java.lang.reflect.Method.invoke(Method.java:564)
>   at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
>   at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
>   at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
>   at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
>   at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
>   at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
>   at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
>   at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
>   at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
>   at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
>   at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
>   at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
>   at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
>   at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
>   at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
>   at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:252)
>   at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:141)
>   at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:112)
>   at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
>   at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
>   at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
>   at java.base/java.lang.reflect.Method.invoke(Method.java:564)
>   at org.apache.maven.surefire.util.ReflectionUtils.invokeMethodWithArray(ReflectionUtils.java:189)
>   at org.apache.maven.surefire.booter.ProviderFactory$ProviderProxy.invoke(ProviderFactory.java:165)
>   at org.apache.maven.surefire.booter.ProviderFactory.invokeProvider(ProviderFactory.java:85)
>   at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:115)
>   at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:75)
>   ```
>   
>   
>   
>   * The test should clean up whatever it changed from the database when finished
> * Merge `master` so that we're sure it works with the current code base
>   That'll also allow us to run the JaCoCo report and check coverage

In my case the test works fine. Master is alredy merged and the test erases the data it uses.

Reply:
Yes, don't worry about the Exception, it was a problem on my end!